frontend/test/e2e/ManageProviders.spec.ts

```diff
@@ -0,0 +1,113 @@
+import { By, until, WebDriver } from 'selenium-webdriver';
+
+import { SettingsPage } from '../pages/SettingsPage.js';
+import { ContractsPage } from '../pages/ContractsPage.js';
+import { before, describe, after, it } from 'node:test';
+import { expect } from 'chai';
+import { getDriver } from '../utils/driver.js';
+
+let driver: WebDriver;
+let settingsPage: SettingsPage;
+let contractsPage: ContractsPage;
+
+const providerParams = {
+  provider: '0-custom-provider',
+  model: '0-custom-model',
+  plugin: 'openai',
+};
+
+describe('Settings - Manage Providers', () => {
+  before(async () => {
+    driver = await getDriver();
+    settingsPage = new SettingsPage(driver);
+    contractsPage = new ContractsPage(driver);
+  });
+
+  it('should create and update a custom provider', async () => {
+    await contractsPage.navigate();
+    await contractsPage.waitUntilVisible();
+    await contractsPage.skipTutorial();
+
+    await settingsPage.navigate();
+    await settingsPage.waitUntilVisible();
+
+    const initialProviders = await settingsPage.getProvidersElements();
+
+    // Create
+    await settingsPage.createCustomProvider(providerParams);
+
+    const newProviders = await settingsPage.getProvidersElements();
+
+    expect(
+      newProviders.length,
+      'number of providers should be greather than old providers list',
+    ).be.greaterThan(initialProviders.length);
```

Comment:
@denishacquin we should be more specific and expecting the exact number of providers after the creation